composer/loggers/mosaicml_logger.py

```diff
@@ -177,20 +180,20 @@ def _flush_metadata(self, force_flush: bool = False) -> None:
         """Flush buffered metadata to MosaicML if enough time has passed since last flush."""
         if self._enabled and (time.time() - self.time_last_logged > self.log_interval or force_flush):
             try:
-                mcli.update_run_metadata(self.run_name, self.buffered_metadata)
+                f = mcli.update_run_metadata(self.run_name, self.buffered_metadata, future=True, protect=True)
```

Comment:
is the retry logic now inside this call?

Reply:
Yes I believe so but I couldn't find where that retry logic was. @siriuslee mind linking me to how the retry logic is implemented?

Reply:
Yes, it's within the the `update_run_metadata` call now. We use:
```
MAX_RETRIES = 10
MAX_INTERVAL = 60
RETRY_EXCEPTIONS = {
    HTTPStatus.TOO_MANY_REQUESTS,
    HTTPStatus.INTERNAL_SERVER_ERROR,
    HTTPStatus.BAD_GATEWAY,
    HTTPStatus.SERVICE_UNAVAILABLE,
    HTTPStatus.GATEWAY_TIMEOUT,
}
backoff.on_exception(
        backoff.expo,
        MAPIException,
        max_tries=MAX_RETRIES,
        jitter=backoff.random_jitter,
        giveup=should_giveup,
        logger=logger,
        backoff_log_level=logging.DEBUG,
        giveup_log_level=logging.DEBUG,
        max_value=MAX_INTERVAL,
    )
```